ntbs-service/Pages/Notifications/Edit/_SaveButton.cshtml

```diff
@@ -5,11 +5,17 @@
 
 @{
     var isDraft = Model.NotificationStatus == NotificationStatus.Draft;
-    var isEditPage = (bool?) ViewData["IsEditPage"] ?? true;
+    var pageHasEditableContent = (bool?) ViewData["PageHasEditableContent"] ?? true;
     var currentPage = (string) ViewData["CurrentPage"];
+    var isEditMDRPage = currentPage == NotificationSubPaths.EditMDRDetails;
+    var isEditTreatmentEventsPage = currentPage == NotificationSubPaths.EditTreatmentEvents;
```

Comment:
I thought the previous separation of concerns was better i.e. passing `IsLastPage` to the save button.

That way the save button has responsibility for how it displays itself, but the pages have responsibility for knowing whether or not they are the last page.

The way that you've written it, the save button has responsibility for both of these things, and I think this has two main disadvantages:

- Conceptually it seems weird that the save button, which is a very low level component, should know so much about high level concepts like the order of pages.
- It makes this view more complicated, and overall it makes the app more difficult to understand.

Reply:
Thanks Josh, I think what I have now is much better